composer/callbacks/mlperf.py

```diff
@@ -224,36 +226,53 @@ def _log_dict(self, data: Dict[str, Any]):
         for key, value in data.items():
             self.mllogger.event(key=key, value=value)
 
-    def _get_accuracy(self, state: State):
+    def _get_accuracy(self, state: State) -> float:
         if self.metric_name not in state.current_metrics[self.metric_label]:
-            raise ValueError('Accuracy must be a validation metric.')
-        return state.current_metrics[self.metric_label][self.metric_name]
+            raise ValueError(f'{self.metric_name} must be a validation metric.')
+
+        metric = state.current_metrics[self.metric_label][self.metric_name]
+        return float(metric)
+
+    def _get_dataloader_stats(self, dataloader: Iterable):
+        """ returns tuple of (batch_size, num_samples)"""
+
+        if isinstance(dataloader, DataLoader):
+            return (dataloader.batch_size, len(dataloader.dataset))  # type: ignore
```

Comment:
That would not be accurate when `drop_last = False`, which is what is required for `eval` though?

Reply:
Fair point; though on that note, len(train_dataloader.dataset) is inaccurate since we usually drop last when training :D.

If this information must be set before training, I don't have a good solution here, other than perhaps asking for it as an init variable and then verifying it against `state.timestamp`?

(Side rant: this would be soo much easier if we could log this _after_ training, since it's accurately tracked in `state.timestamp.sample`.)

Anyways, don't have strong opinions here.

Reply:
The MLPerf requirements for this entry are the number of samples in the dataset, not the number of samples per epoch, which is why `drop_last=False` number is what needs to be reported, regardless of what is being used during training. This is used to check that you indeed are operating on a full copy of ImageNet.